backend/endpoints/login.go

```diff
@@ -0,0 +1,71 @@
+package endpoints
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-kit/kit/endpoint"
+
+	"github.com/dwarvesf/smithy/backend"
+	BackendConfig "github.com/dwarvesf/smithy/backend/config"
+	"github.com/dwarvesf/smithy/backend/service"
+)
+
+// In this version, admin have two role
+const (
+	admin string = "admin"
+	user  string = "user"
+)
+
+// LoginRequest store login structer
+type LoginRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// LoginResponse store login respone
+type LoginResponse struct {
+	Authentication string `json:"authentication"`
+}
+
+func makeLoginEndpoint(s service.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req, ok := request.(LoginRequest)
+
+		if !ok {
+			return nil, errors.New("Login fail")
+		}
+
+		// check username and password is existed in persistant,
+		// if exist, will return jwt with role and username
+		// otherwise return login fail
+
+		// if login fail
+		ok, rule := login(req.Username, req.Password, s.Config.Config().UserConfig.Users)
+
+		if !ok {
+			return nil, errors.New("Login fail")
```

Comment:
We should add custom error for this case because in currently it will have status code 500 (internal server error) it will make front-end complain us